src/exporters/vhi.py

```diff
@@ -0,0 +1,220 @@
+        for raw_filename in legit_filenames:
+            output_filename = (
+                make_filename(raw_folder, raw_filename, dataset='vhi')
+            )
+            download_file_from_ftp(ftp, raw_filename, output_filename)
+
+    return
+
+
+class VHIExporter(BaseExporter):
+    """Exports Vegetation Health Index from NASA site
+
+    ftp.star.nesdis.noaa.gov
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+    @staticmethod
+    def get_ftp_filenames() -> List:
+        """  get the filenames containing VHI """
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # append the filenames to a list
+            listing: List = []
+            ftp.retrlines("LIST", listing.append)
+            # extract the filename
+            filepaths = [f.split(' ')[-1] for f in listing]
+            # extract only the filenames of interest
+            vhi_files = [f for f in filepaths if ".VH.nc" in f]
+
+        return vhi_files
+
+    # @staticmethod
+    def chunks(self, l: List, n: int) -> Generator:
+        # return a generator object which chunks list into sublists of size n
+        # https://chrisalbon.com/python/data_wrangling/break_list_into_chunks_of_equal_size/
+        # For item i in a range that is a length of l,
+        for i in range(0, len(l), n):
+            # Create an index range for l of n items:
+            yield l[i:i + n]
+
+    def save_errors(self, outputs: List) -> None:
+        print("\nError: ", [errors for errors in outputs if errors is not None])
+
+        # save the filenames that failed to a pickle object
+        with open(self.raw_folder / 'vhi_export_errors.pkl', 'wb') as f:
+            pickle.dump([error[-1] for error in outputs if error is not None], f)
+
+        return
+
+    def run_parallel(self,
+                     vhi_files: List,
+                     years: List[int] = [yr for yr in np.arange(1981, 2020)]
+                     ) -> List:
+        pool = Pool(processes=100)
+
+        # split the filenames into batches of 100
+        batches = [batch for batch in self.chunks(vhi_files, 100)]
+
+        # run in parallel for multiple file downloads
+        # passing mutliple args using partial https://stackoverflow.com/a/5442981/9940782
+        args = dict(raw_folder=self.raw_folder, years=years)
+        outputs = pool.map(partial(batch_ftp_request, args), batches)
+
+        # write the output (TODO: turn into logging behaviour)
+        print("\n\n*************************")
+        print("VHI Data Downloaded")
+        print("*************************")
+        print("Errors:")
+        pprint([error for error in outputs if error is not None])
+        # save errors
+        self.save_errors(outputs)
+
+        return batches
+
+    def check_failures(self) -> List:
```

Comment:
This function is never called. Can we get rid of it?

Reply:
I haven't found out where to call it yet - I think that it's still useful because if we want to 'retry' running for the files that have been missed. For example if you run out of space, it will catch the files htat haven't been downloaded and then you can clear space and continue the download?

What do you think?

Reply:
Hmm, that is fair, but the errors will be printed before they are saved. I guess my concern is that having it in this class makes it unlikely users will use it.

Perhaps an alternative is to save it as a text file (so its easier to access), and then (if errors are saved) print at the end of the export `"errors saved in errors.txt"`.

This does highlight the need for logging infrastructure, which can be one of the next things we work on.

Reply:
changed the print statement slightly can you check in the most up to date `vhi.py` argument?

Reply:
Will remove if you think is unnecessary